python/lsst/pipe/tasks/htmIndexEngine.py

```diff
@@ -0,0 +1,352 @@
+#
+# LSST Data Management System
+# Copyright 2008-2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.afw.table as afwTable
+import lsst.afw.coord as afwCoord
+import lsst.afw.geom as afwGeom
+from lsst.afw.image import fluxFromABMag, fluxErrFromABMagErr
+
+import esutil
+import numpy
+
+class IngestIndexedReferenceConfig(pexConfig.Config):
+    ref_dataset_name = pexConfig.Field(
+            dtype = str,
+            default = 'cal_ref_cat',
+            doc = 'String to pass to the butler to retrieve persisted files.',
+    )
+    ra_name = pexConfig.Field(
+            dtype = str,
+            default = '',
```

Comment:
Good point.  I'm forcing people to set these in the validate step, but I'll take out the default.